Code/Mantid/Framework/DataHandling/inc/MantidDataHandling/AppendGeometryToSNSNexus.h

```diff
@@ -77,16 +77,16 @@ class DLLExport AppendGeometryToSNSNexus : public API::Algorithm {
   bool m_makeNexusCopy;
 
   /// Algorithm progress keeper
-  API::Progress *progress;
+  API::Progress *m_progress;
 
```

Comment:
With this m_progress I think we have the same issue as I've seen in other algorithms. Somewhere in the implementation it is allocated with new but then never deleted. This is just a small leak but it could be fixed here, simply adding the missing delete or, most likely better, by using a smart pointer. Oh, and also progress/m_progess does not need to be a member of the class, it can live just inside the exec method().